### ISSUE TITLE
Appium-1.6.4-beta-WebDriverAgent stuck on createSession

### ISSUE THREAD

**Issue by saikrishna321** [appium/appium#7804]
Appium] Welcome to Appium v1.6.4-beta
[Appium] Non-default server args:
[Appium]   address: '127.0.0.1'
[Appium]   port: 60899
[Appium]   sessionOverride: true
[Appium]   log: '/Users/saikrisv/git/new_appium_parallel/AppiumTestDistribution/target/appiumlogs/ffa50fa3ec0764135fb32ab7024a30a929af5093__HomePageTest2.txt'
[Appium]   tmpDir: '/Users/saikrisv/git/new_appium_parallel/AppiumTestDistribution/target/tmp_60899'
[Appium]   webkitDebugProxyPort: 60833
[Appium] Appium REST http interface listener started on 127.0.0.1:60899
[HTTP] --> GET /wd/hub/status {}
[debug] [MJSONWP] Calling AppiumDriver.getStatus() with args: []
[debug] [MJSONWP] Responding to client with driver.getStatus() result: {"build":{"version":"1.6.4-beta","revision":null}}
[HTTP] <-- GET /wd/hub/status 200 14 ms - 88 
Before Class called11
com.test.site.HomePageTest2
Setting iOS Desired Capabilities:
Feb 06, 2017 10:59:33 AM org.openqa.selenium.remote.ProtocolHandshake createSession
INFO: Attempting bi-dialect session, assuming Postel's Law holds true on the remote end
[HTTP] --> POST /wd/hub/session {"capabilities":{"desiredCapabilities":{"app":"/Users/saikrisv/git/VodQaAdvancedAppium/VodQAReactNative.ipa","platformVersion":"10.0","bundleId":"XXXXX","automationName":"XCuiTest","udid":"ffa50fa3ec0764135fb32ab7024a30a929af5093","platformName":"iOS","deviceName":"iPhone","autoAcceptAlerts":true},"requiredCapabilities":{}},"desiredCapabilities":{"app":"/Users/saikrisv/git/VodQaAdvancedAppium/VodQAReactNative.ipa","platformVersion":"10.0","bundleId":"XXXX","automationName":"XCuiTest","udid":"ffa50fa3ec0764135fb32ab7024a30a929af5093","platformName":"iOS","deviceName":"iPhone","autoAcceptAlerts":true},"requiredCapabilities":{}}
[debug] [MJSONWP] Calling AppiumDriver.createSession() with args: [{"app":"/Users/saikrisv/git/VodQaAdvancedAppium/VodQAReactNative.ipa","platformVersion":"10.0","bundleId":"XXXX","automationName":"XCuiTest","udid":"ffa50fa3ec0764135fb32ab7024a30a929af5093","platformName":"iOS","deviceName":"iPhone","autoAcceptAlerts":true},{},{"desiredCapabilities":{"app":"/Users/saikrisv/git/VodQaAdvancedAppium/VodQAReactNative.ipa","platformVersion":"10.0","bundleId":"XXXX","automationName":"XCuiTest","udid":"ffa50fa3ec0764135fb32ab7024a30a929af5093","platformName":"iOS","deviceName":"iPhone","autoAcceptAlerts":true},"requiredCapabilities":{}},null,null]
[Appium] Creating new XCUITestDriver (v2.7.0) session
[Appium] Capabilities:
[Appium]   app: '/Users/saikrisv/git/VodQaAdvancedAppium/VodQAReactNative.ipa'
[Appium]   platformVersion: '10.0'
[Appium]   bundleId: 'XXXXX'
[Appium]   automationName: 'XCuiTest'
[Appium]   udid: 'ffa50fa3ec0764135fb32ab7024a30a929af5093'
[Appium]   platformName: 'iOS'
[Appium]   deviceName: 'iPhone'
[Appium]   autoAcceptAlerts: true
[debug] [XCUITest] XCUITestDriver version: 2.7.0
[XCUITest] The capabilities 'autoAcceptAlerts' and 'autoDismissAlerts' do not work for XCUITest-based tests. Please adjust your alert handling accordingly.
[BaseDriver] Session created with session id: 76bab1ad-a8b9-4240-9b96-b14bd902d5de
[debug] [XCUITest] Xcode version set to '8.2'
[debug] [XCUITest] iOS SDK Version set to '10.2'
[debug] [XCUITest] Available devices: ffa50fa3ec0764135fb32ab7024a30a929af5093
[debug] [XCUITest] Creating iDevice object with udid 'ffa50fa3ec0764135fb32ab7024a30a929af5093'
[XCUITest] Determining device to run tests on: udid: 'ffa50fa3ec0764135fb32ab7024a30a929af5093', real device: true
[BaseDriver] Using local app '/Users/saikrisv/git/VodQaAdvancedAppium/VodQAReactNative.ipa'
[debug] [BaseDriver] Copying local zip to tmp dir
[debug] [BaseDriver] Unzipping /var/folders/yx/q5fqv2j55_ddwj2m5_x8jv7h0000gn/T/201716-8339-1uw6ymc.banufh6w29/appium-app.zip
[debug] [BaseDriver] Testing zip archive: /var/folders/yx/q5fqv2j55_ddwj2m5_x8jv7h0000gn/T/201716-8339-1uw6ymc.banufh6w29/appium-app.zip
[BaseDriver] Unzipped local app to '/var/folders/yx/q5fqv2j55_ddwj2m5_x8jv7h0000gn/T/201716-8339-1uw6ymc.banufh6w29/Payload/VodQAReactNative.app'
[debug] [XCUITest] Checking whether app '/var/folders/yx/q5fqv2j55_ddwj2m5_x8jv7h0000gn/T/201716-8339-1uw6ymc.banufh6w29/Payload/VodQAReactNative.app' is actually present on file system
[debug] [XCUITest] App is present
[debug] [iOSLog] Attempting iOS device log capture via libimobiledevice idevicesyslog
[debug] [iOSLog] Found idevicesyslog: '/usr/local/bin/idevicesyslog'
[debug] [XCUITest] Reset: running ios real device reset flow
[XCUITest] Setting up real device
[debug] [XCUITest] App is not installed. Will try to install.
[debug] [XCUITest] App installed successfully.
[XCUITest] Using WDA path: '/usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent'
[XCUITest] Using WDA agent: '/usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent/WebDriverAgent.xcodeproj'
[XCUITest] Launching WebDriverAgent on the device
[debug] [XCUITest] Carthage found: '/usr/local/bin/carthage'
[debug] [XCUITest] Killing hanging processes
[debug] [XCUITest] Beginning test with command 'xcodebuild build-for-testing test-without-building -project /usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent/WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -destination id=ffa50fa3ec0764135fb32ab7024a30a929af5093 -configuration Debug' in directory '/usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent'
[debug] [XCUITest] Starting iproxy to forward traffic from local port 8100 to device port 8100 over USB
[debug] [XCUITest] Waiting up to 60000ms for WebDriverAgent to start
[debug] [XCUITest] Log file for xcodebuild test: /Users/saikrisv/Library/Developer/Xcode/DerivedData/WebDriverAgent-brdadhpuduowllgivnnvuygpwhzy/Logs/Test/C8065F3B-FCFF-49EC-B101-F9288E73B759/Session-WebDriverAgentRunner-2017-02-06_105941-cgqL3p.log
[debug] [XCUITest] WebDriverAgent successfully started after 9765ms
[debug] [XCUITest] Sending createSession command to WDA
[debug] [JSONWP Proxy] Proxying [POST /session] to [POST http://localhost:8100/session] with body: {"desiredCapabilities":{"bundleId":"XXXXX","arguments":[],"environment":{},"shouldWaitForQuiescence":true}}

**Note::** open the WebDriverAgent app but fails to open the AUT. Was working fine with previous version. 

@mykola-mokhnach @imurchie any help on this pls

**Comment by saikrishna321** [comment]
When running the command manually 

```
xcodebuild build-for-testing test-without-building -project /usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent/WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -destination id=ffa50fa3ec0764135fb32ab7024a30a929af5093 -configuration Debug' in directory '/usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent


~/git/new_appium_parallel/AppiumTestDistribution(branch:master*) » xcodebuild build-for-testing test-without-building -project /usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent/WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -destination id=ffa50fa3ec0764135fb32ab7024a30a929af5093 -configuration Debug' in directory '/usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent
=== BUILD TARGET WebDriverAgentLib OF PROJECT WebDriverAgent WITH THE DEFAULT CONFIGURATION (Release) ===

Check dependencies

=== BUILD TARGET WebDriverAgentRunner OF PROJECT WebDriverAgent WITH THE DEFAULT CONFIGURATION (Release) ===

Check dependencies

** TEST BUILD SUCCEEDED **

2017-02-06 11:09:55.939 xcodebuild[8496:456229]  IDETestOperationsObserverDebug: Writing diagnostic log for test session to:
/Users/saikrisv/Library/Developer/Xcode/DerivedData/WebDriverAgent-brdadhpuduowllgivnnvuygpwhzy/Logs/Test/B180F14E-DB21-47B9-A778-708F1382C9D8/Session-WebDriverAgentRunner-2017-02-06_110955-ImenQW.log
2017-02-06 11:09:55.939 xcodebuild[8496:456191] [MT] IDETestOperationsObserverDebug: (BD6E1A6A-5335-4D12-84B2-4CA8216311D6) Beginning test session WebDriverAgentRunner-BD6E1A6A-5335-4D12-84B2-4CA8216311D6 at 2017-02-06 11:09:55.939 with Xcode 8C38 on target <DVTiOSDevice: 0x7feb3411b960> {
                deviceSerialNumber:         FK1Q1BRXG5QQ
                identifier:                 ffa50fa3ec0764135fb32ab7024a30a929af5093
                deviceClass:                iPhone
                deviceName:                 iPhone
                deviceIdentifier:           ffa50fa3ec0764135fb32ab7024a30a929af5093
                productVersion:             10.1.1
                buildVersion:               14B100
                deviceSoftwareVersion:      10.1.1 (14B100)
                deviceArchitecture:         arm64
                deviceTotalCapacity:        11839356928
                deviceAvailableCapacity:    8182550528
                deviceIsTransient:          NO
                ignored:                    NO
                deviceIsBusy:               NO
                deviceIsActivated:          YES
                deviceActivationState:      Activated
                deviceType:                 <DVTDeviceType:0x7feb31c1fa90 Xcode.DeviceType.iPhone>
                supportedDeviceFamilies:    (
    1
)
                applications:              (null)
                provisioningProfiles:      (null)
                activityProgress:          -2
                activityTitle:             
                hasInternalSupport:        NO
                isSupportedOS:             YES
                developerDiskMountError:   (null)
(null)
        } (10.1.1 (14B100))
writeDictToFile:1278 ==== Successfully wrote Manifest cache to /var/folders/yx/q5fqv2j55_ddwj2m5_x8jv7h0000gn/C/com.apple.DeveloperTools/All/Xcode/EmbeddedAppDeltas/20cb09cbbe3b7f5a073e79366e7d2765/ffa50fa3ec0764135fb32ab7024a30a929af5093/ManifestCache.plist
2017-02-06 11:10:01.589 xcodebuild[8496:456191] Error Domain=IDETestOperationsObserverErrorDomain Code=5 "Early unexpected exit, operation never finished bootstrapping - no restart will be attempted" UserInfo={NSLocalizedDescription=Early unexpected exit, operation never finished bootstrapping - no restart will be attempted}
** TEST EXECUTE FAILED **

```

**Comment by mykola-mokhnach** [comment]
What is in the device log?

**Comment by saikrishna321** [comment]
@mykola-mokhnach when i remove the bundle ID it works, guess i have provided the bundle ID incorrect . Closing as user-side issue.

**Comment by sjr03** [comment]
Hey Guys!
Currently I am working in Appium 1.6.3. I've faced a warning in Webdriver Agent, " The file “ProjectSettings.xcconfig” couldn’t be opened because there is no such file.​" So I have updated the Webdriver Agent from this link "https://github.com/facebook/WebDriverAgent". Now in Webdriver Agent it is throwing the following error " 'openURL:' is deprecated: first deprecated in iOS 10.0 - Please use openURL: options:completionHandler: instead".

Current Working Environment:
Xcode -> 8.2.1
Appium -> 1.6.3 (Also tried in Appium 1.6.4 beta and Appium 1.6.1, the same error is displayed)
MAC OS X -> 10.12.2
iPhone version -> 10.2

Please help on this.

**Comment by imurchie** [comment]
@sjr03  A number of things changed in WebDriverAgent since 1.6.3. Updating it alone is a recipe for headaches, and is beyond the scope of our trying to debug stuff.

I would upgrade to Appium 1.6.4-beta (`npm uninstall -g appium && npm install -g appium@beta`). This includes WDA updates, and works with them.

**Comment by sjr03** [comment]
Thank you @imurchie , I'll try this solution.

**Comment by qual** [comment]
Facing this issue in Appium 1.6.4 GUI version as well. If bundle id is passed incorrectly or if app is not present in the iOS device, Appium gets stuck forever. Ideally, Appium should throw SessionNotCreated/WebDriverException like exception as it throws correctly in Android or rather it was throwing earlier for UIAutomation.
Could someone (@mykola-mokhnach @imurchie) please update if this issue is going to be fixed or is it way beyond Appium server's scope ?

**Comment by imurchie** [comment]
https://github.com/appium/appium-xcuitest-driver/pull/428

**Comment by qual** [comment]
@imurchie Thanks for considering this issue for fixing !

**Comment by imurchie** [comment]
Fixed in `appium-xcuitest-driver@2.28.4`.
